expression/aggregation/descriptor.go

```diff
@@ -345,6 +345,9 @@ func (a *AggFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
 		a.Args[0] = expression.BuildCastFunction(ctx, a.Args[0], tp)
 	}
 	a.RetTp = a.Args[0].GetType()
+	if a.RetTp.Tp == mysql.TypeEnum || a.RetTp.Tp == mysql.TypeSet {
```

Comment:
should we also check `TypeBit`?

Reply:
`TypeBit` has been checked when buildMaxMin and buildFirstRow.
I tried to check it here, but it caused other problems,
I'll try to move the check of `TypeBit` here individually.